synapseclient/models/protocols/json_schema_protocol.py

```diff
@@ -0,0 +1,135 @@
+from typing import TYPE_CHECKING, Generator, Optional, Protocol, Union
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+    from synapseclient.models.mixins import (
+        InvalidJSONSchemaValidation,
+        JSONSchemaBinding,
+        JSONSchemaDerivedKeys,
+        JSONSchemaValidation,
+        JSONSchemaValidationStatistics,
+    )
+
+
+class BaseJSONSchemaProtocol(Protocol):
+    """
+    Mixin class to provide JSON schema functionality.
+    This class is intended to be used with classes that represent Synapse entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the entity.
+    """
+
+    id: Optional[str] = None
+
+    def bind_schema(
+        self,
+        json_schema_uri: str,
+        *,
+        enable_derived_annos: bool = False,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> "JSONSchemaBinding":
+        """
+        Bind a JSON schema to the entity.
+
+        Arguments:
+            json_schema_uri (str): The URI of the JSON schema to bind to the entity.
+            enable_derived_annos (bool, optional): If true, enable derived annotations. Defaults to False.
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
```

Comment:
The unfortunately clunky thing we have to worry about when we create these sync function definitions is that we still want to have examples for the doc pages. The examples you have in the async method are great, they could be reuse here and swapped to remove asyncio.

Reply:
Got it. Thanks for the reminder. I have added the examples.